frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java

```diff
@@ -0,0 +1,87 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getGraphTable;
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getSelectedVcsRepository;
+
+import java.util.Collections;
+
+import com.intellij.ide.util.PsiNavigationSupport;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.fileEditor.impl.NonProjectFileWritingAccessProvider;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Computable;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.gitmachete.frontend.vfs.utils.GitVfsUtils;
+import com.virtuslab.gitmachete.frontend.vfs.utils.MacheteFileType;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class OpenMacheteFileAction extends DumbAwareAction {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    Project project = anActionEvent.getProject();
+
+    var gitDir = getSelectedVcsRepository(anActionEvent).map(GitVfsUtils::getGitDirectory);
+
+    if (gitDir.isDefined()) {
+      var macheteFile = ApplicationManager.getApplication().runWriteAction(new Computable<Option<VirtualFile>>() {
+        @Override
+        public Option<VirtualFile> compute() {
+          return Try.of(
+              () -> Option.of(gitDir.get().findOrCreateChildData(/* requestor */ this, /* name */ MacheteFileType.FILE_NAME)))
+              .onFailure(e -> VcsNotifier.getInstance(project).notifyError(/* title */ "",
```

Comment:
Hmm `onFailure` doesn't suppress the exception, it'll still be thrown from the method... is it the intended behavior?